addons/calendar/models/calendar.py

```diff
@@ -1146,14 +1146,19 @@ def get_recurrent_ids(self, domain, order=None):
             for key in (order or self._order).split(',')
         ))
         def key(record):
+            # we need to deal with undefined fields, as sorted requires an homogeneous iterable
+            def boolean_product(x):
+                if isinstance(x, bool):
+                    return (x, x)
+                return (True, x)
             # first extract the values for each key column (ids need special treatment)
             vals_spec = (
                 (any_id2key(record[name]) if name == 'id' else record[name], desc)
```

Comment:
maybe apply boolean_product directly to the else clause of this?

Reply:
No I mean 
```python
(any_id2key(record[name]) if name == 'id' else boolean_product(record[name]), desc)
```
instead of applying it during the list comprehension.

Reply:
Ok